src/Network_Equivalences-Communication.thy

```diff
@@ -13,6 +13,33 @@ lemma adapted_after_distributor:
   shows "(A \<Rightarrow> Bs) \<guillemotleft> \<E> = A \<guillemotleft> \<E> \<Rightarrow> map (\<lambda>B. B \<guillemotleft> \<E>) Bs"
   sorry
 
+lemma transition_from_distributor:
+  assumes "A \<Rightarrow> Bs \<rightarrow>\<^sub>s\<lparr>\<alpha>\<rparr> Q"
+  obtains n and X
+  where
+    "\<alpha> = A \<triangleright> \<star>\<^bsup>n\<^esup> X"
+  and
+    "Q = \<Prod>B \<leftarrow> Bs. B \<guillemotleft> suffix n \<triangleleft> X \<parallel> A \<guillemotleft> suffix n \<Rightarrow> map (\<lambda>B. B \<guillemotleft> suffix n) Bs"
+proof -
+  obtain n and X
+  where "\<alpha> = A \<triangleright> \<star>\<^bsup>n\<^esup> X" and "Q = post_receive n X (\<lambda>x. \<Prod>B \<leftarrow> Bs. B \<triangleleft> \<box> x) \<parallel> (A \<Rightarrow> Bs) \<guillemotleft> suffix n"
```

Comment:
Also this `where` clause should be indented.

I think we shouldn’t break it into several lines at this point, although after indentation it will slightly exceed the area marked as valid in the IDE. Whether it will ultimately really not fit on a line in the PDF document is not clear, in particular as this depends on layout decisions still to be made.

Reply:
Also fixed in f703d9e471c70cd3db997e0de6ab1ad67e1e66de, please verify.